convert2rhel/pkghandler.py

```diff
@@ -1144,52 +1111,147 @@ def clean_yum_metadata():
 
 
-def _package_version_cmp(pkg_1, pkg_2):
-    """Compare the version key in a given package name.
+def compare_package_versions(version1, version2):
+    """Compare two EVR versions against each other, as well as
+    the name and architectures
 
-    Consider the following variables that will be passed to this function::
 
-        pkg_1 = 'kernel-core-0:4.18.0-240.10.1.el8_3.x86_64'
-        pkg_2 = 'kernel-core-0:4.18.0-239.0.0.el8_3.x86_64'
+    :param version1: The version to be compared.
+    :type version1: str
+    :param version2: The version to compare against.
+    :type version2: str
 
-    The output of this will be a tuple containing the package version in a
-    tuple::
+    :example:
 
-        result = _package_version_cmp(pkg_1, pkg_2)
-        print("Result is: %s" % result)
-        # Result is: -1
+        >>> match = compare_package_versions("kernel-core-5.14.10-300.fc35", "kernel-core-5.14.15-300.fc35")
+        >>> match # -1
 
-    The function will ignore the package name as it is not an important
-    information here and will only care about the version that is tied to it's
-    name.
+    .. note::
 
-    :param pkg_1: The first package to extract the version
-    :type pkg_1: str
-    :param pkg_2: The second package to extract the version
-    :type pkg_2: str
-    :return: An integer resulting in the package comparision
+        Since the return type is a int, this could be difficult to understand
+        the meaning of each number, so here is a quick list of the meaning from
+        every possible number:
+            * -1 if the evr1 is less then evr2 version
+            * 0 if the evr1 is equal evr2 version
+            * 1 if the evr1 is greater than evr2 version
+
+    :return: Return a number indicating if the versions match, are less or greater then.
     :rtype: int
     """
+    # call parse_pkg_string to obtain name, epoch, release, version and arch fields
+    # order of fields: name, epoch, version, release, arch
+    version1_components = parse_pkg_string(version1)
+    version2_components = parse_pkg_string(version2)
+
+    # ensure package names match, error if not
+    if version1_components[0] != version2_components[0]:
+        raise ValueError(("%s does not match %s" % (version1_components[0], version2_components[0])))
+
+    # ensure package arches match, error if not
+    if version1_components[4] != version2_components[4] and all(([version1_components[4]], version2_components[4])):
+        raise ValueError(("%s does not match %s" % (version1_components[4], version2_components[4])))
+
+    # create list containing EVR for comparison
+    evr1 = (version1_components[1], version1_components[2], version1_components[3])
+    evr2 = (version2_components[1], version2_components[2], version2_components[3])
+
+    return rpm.labelCompare(evr1, evr2)
+
+    # rename prev use of compare_package_versions to compare_package_strings
+
+
+def parse_pkg_string(pkg):
```

Comment:
Just a question to get something out of the way. Do you see this function being used in other places as well, and not only `compare_package_versions`? If so, then we can leave it as a public function, otherwise, I would vote to make it internal to the module. 

I believe that your intention here is to make a generic function that could be used elsewhere, correct, @pr-watson?

Reply:
Yes, I feel that it will be used elsewhere to parse package strings. One of the motivations for separating the parsing and comparison function into two functions was to be able to parse a package without also comparing them.